pkg/backup/backup.go

```diff
@@ -156,32 +179,50 @@ type backupContext struct {
 	networkPoliciesBackedUp bool
 }
 
+func (ctx *backupContext) log(msg string, args ...interface{}) {
+	ctx.logger.log(msg, args...)
+}
+
 // Backup backs up the items specified in the Backup, placing them in a gzip-compressed tar file
 // written to data. The finalized api.Backup is written to metadata.
-func (kb *kubernetesBackupper) Backup(backup *api.Backup, data io.Writer) error {
-	gzw := gzip.NewWriter(data)
-	defer gzw.Close()
+func (kb *kubernetesBackupper) Backup(backup *api.Backup, data, log io.Writer) error {
```

Comment:
Slightly picky, but personally seeing `log` makes me assume that is actually the vehicle this func should be using to log it's output.  Took me a few passes to figure it out via the context.  Perhaps `results`?